hera_pspec/grouping.py

```diff
@@ -234,6 +243,17 @@ def average_spectra(uvp_in, blpair_groups=None, time_avg=False,
             if stat not in uvp.stats_array.keys():
                 raise KeyError("error_field \"%s\" not found in stats_array keys." % stat)
 
+    # pre-check for choos average weighted by error bars
+    
+    if error_weights is None:
```

Comment:
this block should be moved up 20 lines, before the creation of `stat_l`

Reply:
It is much better in this way!